react/lesson1.md

```diff
@@ -36,175 +91,431 @@ cd my-app/
 
-Example:
-```jsx
-import React, { Component } from 'react';
-import Organisation from './Organisation';
+These components are usually defined using ES6 classes. So instead of returning the JSX directly, it's returned as part of a `render()` function. Here's how we would define the button component we created above using ES6 classes:
+
+```javascript
+class GameButton extends React.Components {
 
-class App extends Component {
   render() {
-    return (
-      <div className="App">
-        <Organisation
-            name="Refugee Support Network"
-            website="http://www.refugeesupportnetwork.org"
-        />
-      </div>
-    );
+    return <button> Awesome button! </button>;
   }
 }
+```
+
+The above is the exact same component we defined using the function. 
+
+Child or `presentational` components are essentially smaller customisable to the data that they receive but usually do not have the data in themselves. These are often referred to as `dumb` components - because they do not process the data in themselves:
+
+While function components are a lot simpler to read and write, using ES6 classes allows us to create more powerful, smarter components.
+
+# JSX
+
+You might have noticed something weird: we were returning some HTML-like code in our components.
+
+That is called JSX, and it's a compromise to allows us to use the HTML we are so familiar inside our JavaScript components. 
+
+Since it's not exactly HTML, there are some gotchas, of which you'll need to be aware of:
+
+- event handlers are capitalized
+
+HTML: `<button onclick="myFunction()"/>` 
+JSX: `button onClick="myFunction()"/>`
+
+- `className` instead of `class`
 
-export default App;
+HTML: `<button class="myButton"/>`
+JSX: `<button className="myButton"/>`
 
+- must always have one child element
+
+Won't work:
+
+```javascript
+return (
+  <p> Hello! </p>
+  <p> This is my blog. </p>
+);
 ```
 
-Child or `presentational` components are essentially smaller customisable to the data that they receive but usually do not have the data in themselves. These are often referred to as `dumb` components - because they do not process the data in themselves:
+Will work:
 
-```jsx
-import React, { Component } from 'react';
+```javascript
+return (
+  <div>
+    <p> Hello! </p>
+    <p> This is my blog. </p>
+  </div>
+);
+```
 
-class Organisation extends Component {
-  render() {
-    const name = this.props.name;
-    const website = this.props.website;
-    return (
-        <div>
-            Name: {name} <br />
-            Website: {website}
-        </div>
-    );
+## Thinking in React
+
+Let's go over some examples and discuss how we would break everything up into smart and dumb components.
+
+# Event handlers
+
+Time to handle a few user events. We'll add a "difficulty setting" input. For now, whenever the user changes that input, we will `console.log` the result.
+
+```javascript
+
+difficultyChange = (event) => {
```

Comment:
I got a bit confused about what this function would be used for until much later on. Could it be named to something that conveys how the UI will change, for example `changeNumberOfButtons`?

Reply:
And `this.state.difficulty` be changed to `this.state.numberOfButtons` (or `buttonCount` or whatever).

Reply:
That makes sense, it's a lot more obvious